feat(header): create Header component to serve as navigation on each page

hannahyockel opened the pull request:
1. the logo is a SVG
2. the logo is a link that points to the home page
3. How it works and About links point to the corresponding sections on the home page via URL hashes #how-it-works and #about
4. Search link points to the search page with javascript as the default value in the URL

hannahyockel committed:
feat(core): create app skeleton

> 1. Install fonts Bitter and Monsterrat > 2. Add CSS reset (normalize.css > 3. Add global styles > 4. Add React Router with Routes '/' and '/search' > 5. Setup initial folder structure and import marketing page images

hannahyockel committed:
feat(router): search link points to the search page with javascript as the default value in the URL

hannahyockel committed:
change nav tag to header tag and add default url param: javascript

hannahyockel committed:
Fix default url param

mentor-tara commented:
Great job :rocket: I'm a bit busy at the moment but I'll leave a review soon.

In the meantime, consider starting with the next task already to practice the professional Git workflow. [If you need a refresher, feel free to follow this section in the course again.](https://profy.dev/project/reddit-timer/while-waiting)[](ckma5leps3kko0b87rxtjjp24)

mentor-tara reviewed (changes requested):
Before I approve this PR I'd like to share a few tips with you.

**Please go through the list below and review your own code first.** Imagine you're the reviewer or a person who didn't see the code before. Ask yourself if everything is clearly written and understandable. Are components and variables properly named?

Reviewing your own code with this mindset is an important practice that will help you learn a lot and write high-quality code.

To make it a bit easier for you here are the tips I promised. Some have a link :link: attached where you can find more information. I'd recommend reading the **bold tips** in any case. Feel free to check the boxes when you think your code fits.

**General:**
- [x] Use human readable URLs [:link:](https://profy.dev/project/reddit-timer/review-tips/human-readable-links)
- [x] Use absolute instead of relative links: `<Link to="/search" />` [:link:](https://profy.dev/project/reddit-timer/review-tips/use-absolute-links)
- [x] No components that solely copy props to their children [:link:](https://profy.dev/project/reddit-timer/review-tips/unnecessary-components)
- [x] Optimize svg files with SVGOMG [:link:](https://profy.dev/project/reddit-timer/review-tips/optimize-svg-with-svgomg)
- [x] Use react-router Link instead of HTML anchor tag [:link:](https://profy.dev/project/reddit-timer/review-tips/link-instead-of-anchor-tag)

**Styles:**
- [x] **Where to put your styles** [:link:](https://profy.dev/project/reddit-timer/review-tips/where-to-put-styles)
- [x] **Refrain from styling children with CSS selectors** [:link:](https://profy.dev/project/reddit-timer/review-tips/css-selectors)
- [x] **Extract shared CSS values** [:link:](https://profy.dev/project/reddit-timer/review-tips/common-css-values-in-theme)
- [x] Be careful with pixel-perfectness when using relative font sizes like `0.9em` [:link:](https://profy.dev/project/reddit-timer/review-tips/careful-with-relative-font-size)

**Tests:**
- [x] **Use React Testing Library for testing** [:link:](https://profy.dev/project/reddit-timer/review-tips/use-react-testing-library)
- [x] Don't write snapshot tests [:link:](https://profy.dev/project/reddit-timer/review-tips/no-snapshot-tests)
- [x] **Prefer integration tests over unit tests** [:link:](https://profy.dev/project/reddit-timer/review-tips/no-unit-tests)
- [x] Get the logo by role with `getByRole` [:link:](https://profy.dev/project/reddit-timer/review-tips/get-logo-by-role)
- [x] How to test hash links [:link:](https://profy.dev/project/reddit-timer/review-tips/testing-hash-links)

Once you're done with reviewing your own code feel free to check the [example implementation](https://profy.dev/project/reddit-timer/code-examples/header-ae7e45f0-51b2-433a-b293-3bef0ae80cce). **But remember: It's an important exercise to review your code first on your own.**

When you think this PR is ready to be merged just let me know by requesting another review. I'll approve right away.[](ckmaavvyw3m7p0b358cxn7dzq)

hannahyockel committed:
Restructures the header component's HTML

hannahyockel committed:
Adds skeleton of Header styled components

hannahyockel committed:
Moves InfoSection from App to home/info-section

hannahyockel committed:
Adds skeleton files for style/ theme and global style.

Adds proper packages to App

Adds theme provider, global style, header, switch and its routes (search and home) to App

hannahyockel committed:
Fixes import/export defaults in App (App, Header, GlobalStyles fixed here)

hannahyockel committed:
(style) Adds body variables to global style and theme

hannahyockel committed:
(style) Completes basic header styling

hannahyockel committed:
(tests) Integrates setup and header tests. Current: 1 passing 3 failing

hannahyockel committed:
(tests) Passes header tests with warnings - Matched leaf route at location .. does not have an element

mentor-tara reviewed (approved):
Great work! :tada::tada::tada: You can merge the PR now.[](ckmabg2tk3mkv0b35j06077kr)

hannahyockel committed:
Merge branch 'main' into header
